pkg/readiness/objset.go

```diff
@@ -27,9 +27,39 @@ type objKey struct {
 	namespacedName types.NamespacedName
 }
 
-type objSet map[objKey]struct{}
-
 func (k objKey) String() string {
 	return fmt.Sprintf("%s [%s]", k.namespacedName.String(), k.gvk.String())
+}
+
+type objData struct {
+	retries int
+}
+
+type objDataMutator func(*objData) *objData
+
+func setRetriesFromFlag(o *objData) *objData {
+	o.retries = *readinessRetries
+	return o
+}
+
+func mutatedObjData(muts ...objDataMutator) objData {
+	out := objData{}
+	for _, m := range muts {
+		m(&out)
+	}
+	return out
+}
+
+type objSet map[objKey]objData
 
+func (o objSet) decrementOrDelete(key objKey) bool {
```

Comment:
What if these methods were on objData instead? Would that give any benefits to the consumer code? Greater uniformity?

Reply:
This and the "Should we just return..." comment are related.  Let's discuss here.

I think it would be good to make this a method of `objData`, since that's what it's really about.  This makes it logical to move the `delete` call _out_ of this function, as the `objData` does not have access to the map it is part of.

I think this is a better separation of concerns, and makes it easier to understand the `objectTracker` without having to jump to the `objData` code.

Reply:
+1, it looks much nicer like this!